src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/SystemApiController.java

```diff
@@ -215,4 +215,31 @@ public SystemApiController(InternalTlsCertificateService internalTlsCertificateS
             throw new InternalServerErrorException(new ErrorDeviation(ANCHOR_FILE_NOT_FOUND));
         }
     }
+
+    @Override
+    @PreAuthorize("hasAuthority('UPLOAD_ANCHOR')")
+    public ResponseEntity<Void> uploadAnchor(Resource anchorResource) {
+        byte[] anchorBytes = ResourceUtils.springResourceToBytesOrThrowBadRequest(anchorResource);
+        try {
+            systemService.uploadAnchor(anchorBytes);
+        } catch (SystemService.InvalidAnchorInstanceException e) {
+            throw new BadRequestException(e);
+        } catch (SystemService.AnchorUploadException e) {
+            throw new InternalServerErrorException(e);
+        }
+        return ApiUtil.createCreatedResponse("/api/system/anchor", null);
+    }
+
+    @Override
+    @PreAuthorize("hasAuthority('UPLOAD_ANCHOR')")
+    public ResponseEntity<Anchor> previewAnchor(Resource anchorResource) {
+        byte[] anchorBytes = ResourceUtils.springResourceToBytesOrThrowBadRequest(anchorResource);
+        AnchorFile anchorFile = null;
+        try {
+            anchorFile = systemService.getAnchorFileFromBytes(anchorBytes);
+        } catch (SystemService.InvalidAnchorInstanceException e) {
```

Comment:
It is a `CodedException` which will be translated to a 500:
```javascript
{
    "status": 500,
    "error": {
        "code": "core.MalformedGlobalConf",
        "metadata": [
            "Content is not allowed in prolog."
        ]
    }
}
```
This should be handled so that a 400 is returned instead.

Reply:
`core.MalformedGlobalConf` is thrown if the schema validation fails. I would like to add (well I did) an exception named `MalformedAnchorException` because it is not so confusing as `MalformedGlobalConf`. At least in this case where one tries to upload a configuration anchor.

Reply:
Great....and that will now be http 400?

Reply:
Yes it is now considered a 400